tools/deploy-static.sh

```diff
@@ -1,7 +1,18 @@
+# Script to prepare Docs for Cloudflare Pages deployment 
+
 #!/usr/bin/env bash
 dirname=$(dirname $0)
 cd $dirname/..
 
+# Build folder is uploaded to Cloudflare
 cp -r static/* build/
-cp config/redirects.txt build/_redirects
-cp build/404/index.html build/404.html
\ No newline at end of file
+
+# Extract only the Cloudflare Redirects from redirect.txt
+DIVISION_LINE=`awk '/####*/ {print NR}' config/redirects.txt | sed -n 3p`
+head -n $DIVISION_LINE config/redirects.txt > build/_redirects
+
+# Command to create redirects that don't have trailing slash since Cloudflare has limited regex support
+# awk '/^\/[[:alnum:]]+/ {print $1}' build/_redirects | sed -e 's!.$!!'
+
+# Copy the 404 page to root of the build folder for CF pages to find it
+cp build/404/index.html build/404.html
```

Comment:
Suggest we re-write this file to verify that the redirects exists with both / and no slash. Assuming we can figure out why CF is treating the redirects as dynamic that is.

```bash
#!/usr/bin/env bash

# Script to prepare Docs for Cloudflare Pages deployment

set -e

dirname=$(dirname $0)
cd "${dirname}"/.. || exit 1

# Build folder is uploaded to Cloudflare
cp -r static/* build/

# normalize all Cloudflare Style Redirects to include both trailing slash and no trailing slash variants
generated="$(sed -r 's/^(\/[^ ]+\/[^\/ ]+) /\1\/ /g' config/redirects.txt | sed -rn 'p; s/^(\/[^ ]+)\/ /\1 /p' | sort)"
current="$(sort config/redirects.txt)"

if [ "${current}" != "${generated}" ]
then
    echo "Error: one or more redirects is missing the trailing-slash or non-trailing slash variant, both must exist"
    diff  <(echo "${current}" ) <(echo "${generated}")
    exit 1
fi

# copy the Cloudflare Style Redirects
sed -rn '/^(\/[^ ]+\/[^\/ ]+)/p' config/redirects.txt build/_redirects

# Copy the 404 page to root of the build folder for CF pages to find it
cp build/404/index.html build/404.html
```
The assumption here is that we commit the redirects.txt with both variations so there are no surprises on deploy.

Reply:
Right on, I agreed with your suggestion and actually added the changes as well locally along with couple of more commits but haven't pushed. I was hoping for Cloudflare to respond sooner but they are taking ample time with it. Fingers crossed for us to get unblocked.

Reply:
Update: Let's do the validation in a separate PR. Pushing without the checks.